cd_scripts/e2e_test.sh

```diff
@@ -0,0 +1,118 @@
+#! /bin/bash
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Print commands and their arguments as they are executed.
+set -x
+
+#details.txt file contains the release version and commit hash of the current release.
+gsutil cp  gs://gcsfuse-release-packages/version-detail/details.txt .
+# Writing VM instance name to details.txt (Format: release-test-<os-name>)
+curl http://metadata.google.internal/computeMetadata/v1/instance/name -H "Metadata-Flavor: Google" >> details.txt
+
+# Based on the os type(from vm instance name) in detail.txt, run the following commands to add starterscriptuser
+if grep -q ubuntu details.txt || grep -q debian details.txt;
+then
+#  For ubuntu and debian os
+    sudo adduser --ingroup google-sudoers --disabled-password --home=/home/starterscriptuser --gecos "" starterscriptuser
+else
+#  For rhel and centos
+    sudo adduser -g google-sudoers --home-dir=/home/starterscriptuser starterscriptuser
+fi
+
+# Run the following as starterscriptuser
+sudo -u starterscriptuser bash -c '
+# Exit immediately if a command exits with a non-zero status.
+set -e
+# Print commands and their arguments as they are executed.
+set -x
+
+#Copy details.txt to starterscriptuser home directory and create logs.txt
+cd ~/
+cp /details.txt .
```

Comment:
nit: I think we can directly use /details.txt instead of copying because anyways you are referencing details.txt like ~/details.txt ?

Reply:
I feel having it in the same directory will be helpful for debugging purpose in future (let's say someone SSH into the VM to debug) as we will have all the files at the same place (starterscriptuser home directory).